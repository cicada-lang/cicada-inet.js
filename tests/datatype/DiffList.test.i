require "DiffList.i"
require "Trivial.i"

claim one_two_soles -- Trivial DiffList end

define one_two_soles
  sole (cons :tail) diff @rot @rot @connect
  sole (cons :tail) sole cons diff @rot @rot @connect
  diff_append
end

one_two_soles @inspect @run @inspect

claim two_two_soles -- Trivial DiffList end

define two_two_soles
  (diff) @spread $front sole cons sole cons front @connect
  (diff) @spread $front sole cons sole cons front @connect
  diff_append
end

two_two_soles @inspect @run @inspect
