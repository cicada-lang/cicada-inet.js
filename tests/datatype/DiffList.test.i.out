net_from_port (diff_append₂)-return
  (diff_append₂)-target!value-(diff₆)
  (diff_append₂)-rest value-(diff₅)
  (diff₆)-front value-(cons₇)
  (diff₆)-back tail-(cons₆)
  (cons₇)-head value-(sole₅)
  (cons₇)-tail value-(cons₆)
  (cons₆)-head value-(sole₄)
  (diff₅)-front value-(cons₅)
  (diff₅)-back tail-(cons₅)
  (cons₅)-head value-(sole₃)
end
net_from_port (diff₇)-value!
  (diff₇)-front value-(cons₇)
  (diff₇)-back tail-(cons₅)
  (cons₇)-head value-(sole₅)
  (cons₇)-tail value-(cons₆)
  (cons₆)-head value-(sole₄)
  (cons₆)-tail value-(cons₅)
  (cons₅)-head value-(sole₃)
end
net_from_port (diff_append₄)-return
  (diff_append₄)-target!value-(diff₁₁)
  (diff_append₄)-rest value-(diff₁₀)
  (diff₁₁)-front value-(cons₁₅)
  (diff₁₁)-back tail-(cons₁₄)
  (cons₁₅)-head value-(sole₁₃)
  (cons₁₅)-tail value-(cons₁₄)
  (cons₁₄)-head value-(sole₁₂)
  (diff₁₀)-front value-(cons₁₃)
  (diff₁₀)-back tail-(cons₁₂)
  (cons₁₃)-head value-(sole₁₁)
  (cons₁₃)-tail value-(cons₁₂)
  (cons₁₂)-head value-(sole₁₀)
end
net_from_port (diff₁₂)-value!
  (diff₁₂)-front value-(cons₁₅)
  (diff₁₂)-back tail-(cons₁₂)
  (cons₁₅)-head value-(sole₁₃)
  (cons₁₅)-tail value-(cons₁₄)
  (cons₁₄)-head value-(sole₁₂)
  (cons₁₄)-tail value-(cons₁₃)
  (cons₁₃)-head value-(sole₁₁)
  (cons₁₃)-tail value-(cons₁₂)
  (cons₁₂)-head value-(sole₁₀)
end
